src/middleware/json-api/_deserialize.js

```diff
@@ -1,35 +1,65 @@
 const _ = require('lodash')
 
-function collection (items, included) {
+const cache = new class {
+  constructor () { this._cache = [] }
+
+  set (type, id, deserializedData) {
+    this._cache.push({
+      type: type,
+      id: id,
+      deserialized: deserializedData
+    })
+  }
+
+  get (type, id) {
+    const match = _.find(this._cache, r => r.type === type && r.id === id)
+    return match && match.deserialized
+  }
+}
+
+function collection (items, included, useCache = false) {
   return items.map(item => {
-    return resource.call(this, item, included)
+    return resource.call(this, item, included, useCache)
   })
 }
 
-function resource (item, included) {
+function resource (item, included, responseModel, useCache = false) {
```

Comment:
Is `responseModel` being used for anything? I think we should get rid of it. I can't find any references in the code to this variable anywhere.